src/lib.rs

```diff
@@ -96,12 +96,11 @@ impl Heap {
     /// enough. The runtime is in O(n) where n is the number of free blocks, but it should be
     /// reasonably fast for small allocations.
     pub fn allocate_first_fit(&mut self, layout: Layout) -> Result<NonNull<u8>, ()> {
-        let aligned_layout = HoleList::align_layout(layout);
-        let res = self.holes.allocate_first_fit(aligned_layout);
+        let res = self.holes.allocate_first_fit(layout);
         if res.is_ok() {
-            self.used += aligned_layout.size();
+            self.used += res.unwrap().1.size();
         }
-        res
+        res.map(|tuple| tuple.0)
```

Comment:
You could simplify this with a `match` to avoid the `unwrap()`:

```rust
match self.holes.allocate_first_fit(layout) {
    Ok((ptr, aligned_layout)) => {
        self.used += aligned_layout.size();
        Ok(ptr)
    }
    Err(err) => Err(err),
}
```